File: DarkOrbit 10.0/Net/netty/commands/UpdateMenuItemCooldownGroupTimerCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class UpdateMenuItemCooldownGroupTimerCommand
    {
        public const short ID = 2072;

        public static byte[] write(CooldownTypeModule cooldownType, ClientUISlotBarCategoryItemTimerStateModule timerState, 
                                    long time, long maxTime)
        {
            var param1 = new ByteArray(ID);
            param1.write(timerState.write());
            param1.write(cooldownType.write());
            param1.writeDouble(time);
            param1.writeDouble(maxTime);
            param1.writeShort(-3269);
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/UpdateMenuItemCooldownGroupTimerCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

/**
 Created by Pedro on 04-04-2015.
 */
public class UpdateMenuItemCooldownGroupTimerCommand
        implements ServerCommand {

    public CooldownTypeModule                          var_2679;
    public long                                        maxTime;
    public long                                        time;
    public ClientUISlotBarCategoryItemTimerStateModule timerState;

    public UpdateMenuItemCooldownGroupTimerCommand(CooldownTypeModule param1,
                                                   ClientUISlotBarCategoryItemTimerStateModule param2, long param3,
                                                   long param4) {
        this.var_2679 = param1;
        this.timerState = param2;
        this.time = param3;
        this.maxTime = param4;
    }

    public static int ID = 413;

    public void write(DataOutputStream out) {
        try {
            out.writeShort(ID);
            this.writeInternal(out);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        {
            try {
                this.var_2679.write(param1);
                param1.writeDouble(this.maxTime);
                param1.writeShort(16635);
                param1.writeDouble(this.time);
                this.timerState.write(param1);
            } catch (IOException e) {
            }
        }
    }
}
